pkgs/development/python-modules/slack-bolt/default.nix

```
@@ -0,0 +1,102 @@
+{ lib
+, stdenv
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, pytestCheckHook
+, boddle
+, cherrypy
+, django
+, falcon
+, fastapi
+, flask
+, flask-sockets
+, moto
+, pyramid
+, pytest-asyncio
+, sanic
+, sanic-testing
+, slack-sdk
+, starlette
+, tornado
+, bottle
+, chalice
+, mock
+}:
+
+buildPythonPackage rec {
+  pname = "slack-bolt";
+  version = "1.16.1";
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "slackapi";
+    repo = "bolt-python";
+    rev = "v${version}";
+    sha256 = "sha256-xzEsTx+44ae++WxArxO2rhCsot+ELuhc0aZ44MXY0c4=";
+  };
+
+  propagatedBuildInputs = [ slack-sdk ];
+
```

Use `optional-dependencies` for extra requirements.

```suggestion

  passthru.optional-dependencies = {
    async = [
      aiohttp
      websockets
    ];
   adapter  = [
      boto3
      ...
    ];
  };

```

And then for the check inputs: 

```nix
  checkInputs = [
    pytestCheckHook
    ...
  ] ++ passthru.optional-dependencies.adapter;
```